### ISSUE TITLE
Restructuring Documents

### ISSUE THREAD

**Issue by kazk** [codewars/docs#172]
We should restructure `content/` directory by topics. I know many of you work on GitHub and it's difficult to make these changes, so I'll do it once we're ready.

---

I realized README is misleading:

> - `/tutorials/`: All docs with kind `tutorial`
> - `/recipes/`: All docs with kind `recipe`
> - `/references/`: All docs with kind `reference`

What I meant was `/tutorials/` path shows all docs with kind `tutorial`, and not to put all docs with kind `tutorial` under `/tutorials/`.

We don't need folders for each kind at the top level because each document has `kind` defined. It's much better to keep documents of the same topic close to each other.

Also, we should think about having natural and consistent URL structure. Deeper paths should have more specific content.

Some examples:

- `/getting-started/`
- `/training/` for anything related to solving kata
  - `/debugging-tips/` troubleshooting
  - `/tutorials/` like how to use sample tests
- `/authoring/` for anything related to authoring kata 
  - `/guidelines/`
  - `/tutorials/` for grouping kinds within the topic
  - `/translating/` for anything related to translating kata
    - `/guidelines/`
  - `/reviewing/`
    - `/guidelines/`
- `/commenting/`, `/communicating/`
- `/languages/` for information about supported languages
- `/community/`

We can have some top level folders for general topics if it doesn't fit anywhere.

- `/concepts/`
- `/references/`

Please let me know if you have any feedback. I'm not sure how we should name the top groups to appear more consistent.

We don't need to do this right now, but thinking about the structure might be helpful when writing docs.

**Comment by hobovsky** [comment]
The main problem I have is that I simply do not understand how all these things like file paths, tags, kinds, etc. translate into URLs. Path translates to URL in very natural way for me, and I did not get to wrapping my head around other mechanisms available in our docs.

The other thing is my, probably far from perfect, method of creating many parallel branches and PRs. If it stops you from doing some changes, then you do't have to worry for that. Just tell me and I can always delete some branch or draft PR and get it back later. No big issue for me. I need to rework this part of my working method in this anyway.

**Comment by kazk** [comment]
> The main problem I have is that I simply do not understand how all these things like file paths, tags, kinds, etc. translate into URLs.

For Markdown files in `content/`, URLs are based on their relative paths from `/content` minus the file extension. We should always use `-` for filenames with multiple words because I think Gridsome (the framework we're using) transforms `_` to `-` and that can be confusing. Also, when referring to other pages from a document, always include a trailing slash.
One exception is `index.md` files, those are resolved with the name of the directory like how `index.html` often does.

- `content/foo/bar.md` is `/foo/bar/`
- `content/foo/index.md` is `/foo/`
- `content/foo-bar.md` is `/foo-bar/`

Anything else are generated from some data.

- `/tutorials/` lists links to all documents using metadata `kind: tutorial`. No need to have files under `content/tutorials/` to be listed there.
- `/tags/` lists all the tags we have used
- `/tags/:tag/` lists links to all documents with `tag`
- `/languages/:language/` are generated from language data in `data/languages/`
- `/glossary/` is generated from `data/glossary.yml`

Collections of documents like `/tutorials/` are there for convenience and you shouldn't need to think about them when writing documents. The goal of those collections are to use some metadata to organize the documents in some convenient ways. It's currently not useful, but we can later group those pages in more useful ways with additional metadata.

---

> There are many axes to cut through the docs and I still did not understand how these axes are expressed in the docs system. Which one is tags, which one is kinds, which is file system, which is topics, etc.

File system should be considered first because that's what the URLs are based on and it's permanent. Anything else (metadata) can be used to provide more convenient presentation when needed.

`kind` should follow this idea:

![image](https://user-images.githubusercontent.com/639336/99134370-41122480-25d2-11eb-88d7-3684c444d1c0.png)

Each document should focus on one kind, so there's a clear purpose and target audience. It's annoying to read when these are mixed.

The terms used in our docs are defined in https://github.com/codewars/docs/blob/06dc6dcc9f48530e3046c472aea523c1d5bc5372/data/kinds.yml#L9-L37

For example, I'd classify a guideline as a `reference` and not a `recipe` because it "describes" the requirements.

---

> Another thing is that I am not sure if a set of pages can be effectively collected along more than one axis. Will things like navigations, titles, content flow, align properly when you collect one page (for example, "translations authoring guidelines") once in a set of pages related to "guidelines" (kata, translations, descriptions, ...), and another time in a set of pages related to "translations" (explanation, authoring guidelines, manual, FAQ).

I'm not sure what you have in mind, but we can do pretty much anything. We'll add any necessary metadata, query & group the documents, and present the result.

**Comment by hobovsky** [comment]
> For example, I'd classify a guideline as a reference and not a recipe because it "describes" the requirements.

The fact that I marked guidelines as recipes is that the initial idea I had was somewhat different, and when I was thinking a bit then I split into two parts. One of them would be "functional howto" (set of recipes), and another part would be guidelines we have now. I did not think they could be marked as reference, but now I think about it, I believe they can.

I will try and think on organization when I merge all PRs and branches for guidelines I have pending. I will not create new ones if I'd think they will take a long time to complete.

I will try and digest your above explanation and see what I can come up with. But if you'd like to reorganize something and move some pages, then go ahead.
